drivers: stm32: tree-wide: DT clock helper macros + clean indentation in instance definitions macros

etienne-lms opened the pull request:
Add few `STM32_CLOCK_*()` and `STM32_DT_INST_*()` helper macros to get all clock information from the devicetree. Change all applicable STM32 drivers to uses them or the already existing macros to ensure all clock information are properly set in clock consumer device configuration data.

Prior these changes some drivers only got the bus and bits position information and missed the clock division information which is fine only when this division information is 0.

By the way, cleanup indentation in the macros that were modified to use `STM32_*_CLOCK*()` macros.

etienne-lms committed:
include: drivers: clock_control: stm32: remove deprecated macro

Remove STM32_DT_DEV_DOMAIN_CLOCK_SUPPORT macro that is unused and
not planned to be.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms committed:
drivers: stm32: use STM32_CLOCK_INFO_BY_NAME() and friends

Use STM32_CLOCK_INFO(), STM32_DT_INST_CLOCK_INFO(),
STM32_CLOCK_INFO_BY_NAME() and STM32_DT_INST_CLOCK_INFO_BY_NAME()
helper macros in STM32 drivers.

Using these macros ensure the clock division factor is properly
populated according to DT information. Prior these changes some
drivers only got the bus and bits position information and missed
the clock division information which is fine only when this division
factor information is 0.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms committed:
drivers: dac: stm32: clean indentation in instance init macros

Clean indentation in STM32_DAC_INIT() macro.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms committed:
drivers: can: stm32: clean indentation in instance init macros

Clean indentation in macros used to define device instances in STM32 CAN
driver.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms committed:
drivers: dma: stm32: clean indentation in instance init macros

Clean indentation in macros used to define DMA instances in STM32
DMA drivers.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms committed:
drivers: sensor: stm32_digi_temp: clean indentation in init macros

Clean indentation in STM32_DIGI_TEMP_INIT() macro.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms committed:
drivers: misc: stm32n6_axisram: clean instance init macro indentation

Clean indentation in STM32N6_AXISRAM_INIT() local macro to better
highlight what the macro does. To make it more readable, add a
STM32N6_AXISRAM_MAYBE_INIT() helper macro and use IF_ENABLED()
instead of COND_CODE_0().

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms committed:
include: drivers: clock_control: stm32: add DT clock helper macros

Add helper macros to get clock information from the devicetree:
STM32_DT_INST_CLOCK_INFO_BY_INDEX(), STM32_DT_INST_CLOCK_INFO(),
STM32_CLOCK_INFO_BY_NAME() and STM32_DT_INST_CLOCK_INFO_BY_NAME().

Using these macros or the existing STM32_CLOCK_INFO(), STM32_DT_CLOCKS()
and STM32_DT_INST_CLOCKS() macros ensure clock consumer gets all the
required STM32 clock information that are abstracted by the
STM32 clock drivers resources.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

gautierg-st commented on drivers/adc/adc_stm32.c:
```diff
@@ -1954,53 +1954,53 @@ DT_INST_FOREACH_STATUS_OKAY(GENERATE_ISR)
+		.has_deep_powerdown = DT_INST_PROP(index, st_adc_has_deep_powerdown),	\
+		.has_channel_preselection = DT_INST_PROP(index, st_adc_has_channel_preselection), \
+		.has_differential_support = DT_INST_PROP(index, st_adc_has_differential_support), \
+		.sampling_time_table = DT_INST_PROP(index, sampling_times),		\
```
Align all `\` while at it?

gautierg-st commented on drivers/dma/dma_stm32.c:
```diff
@@ -716,30 +716,28 @@ static DEVICE_API(dma, dma_funcs) = {
-		    &dma_stm32_data_##index, &dma_stm32_config_##index,	\
-		    PRE_KERNEL_1, CONFIG_DMA_INIT_PRIORITY,		\
-		    &dma_funcs)
+	const struct dma_stm32_config dma_stm32_config_##index = {		\
```
Align the `\`

gautierg-st commented on drivers/ipm/ipm_stm32_hsem.c:
```diff
@@ -203,15 +203,16 @@ static const struct stm32_hsem_mailbox_config stm32_hsem_mailbox_0_config = {
-				&stm32_hsem_mailbox_ipm_dirver_api);	\
+				POST_KERNEL,					\
+				CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,		\
+				&stm32_hsem_mailbox_ipm_dirver_api);		\
```
Align with `(`

etienne-lms replied:
ok. Actually, I'll remove this commit from the series. The indentation fixes here are not related to clock info changes from the first commits. I will propose it in another P-R, along equivalent changes in other STM32 drivers.

gautierg-st reviewed (commented):
One mistake to fix, and lots of nitpick on alignment

erwango reviewed (changes requested):
Skipping the indentation stuff

etienne-lms commented on drivers/adc/adc_stm32.c:
```diff
@@ -1963,19 +1963,13 @@ PINCTRL_DT_INST_DEFINE(index);						\
-			.enr = DT_INST_CLOCKS_CELL_BY_NAME(index, adc_pre, bits)},		\
-	 .has_pclken_pre = true,),								\
-	(.has_pclken_pre = false,))								\
-	.clk_prescaler = ADC_STM32_DT_PRESC(index),			\
```
My mistake. Sorry, I message up when squashing the several changes.
Thanks.

etienne-lms replied:
Addressed.

etienne-lms committed:
drivers: adc: stm32: clean instance init macro indentation

Cleanup indentation in ADC_STM32_INIT() macro.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms commented on include/zephyr/drivers/clock_control/stm32_clock_control.h:
Thanks for the suggestion!

etienne-lms commented on drivers/ethernet/eth_stm32_hal_common.c:
```diff
@@ -422,15 +422,11 @@ PINCTRL_DT_INST_DEFINE(0);
-		      .enr = DT_INST_CLOCKS_CELL_BY_NAME(0, mac_clk_rx, bits)},
+	.pclken = STM32_CLOCK_INFO_BY_NAME(DT_INST_PARENT(0), stm_eth),
+	.pclken_tx = STM32_CLOCK_INFO_BY_NAME(DT_INST_PARENT(0), mac_clk_tx),
+	.pclken_rx = STM32_CLOCK_INFO_BY_NAME(DT_INST_PARENT(0), mac_clk_rx),
```
> only stm_eth is on the parent, the others aren't.

Right, thanks.

> also maybe take a look at my comment #96134 (comment)

I guess that's something to consider. Would you be okay we address this in a specific P-R, focusing on stm32/eth clocks?

etienne-lms committed:
include: drivers: clock_control: stm32: document clock macros

Add a bit of documentation for the existing STM32_CLOCK_INFO()
STM32_DT_CLOCKS().

While at it, fix indentation of STM32_CLOCK_INFO() macro.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms commented on include/zephyr/drivers/clock_control/stm32_clock_control.h:
Addressed

etienne-lms commented on drivers/mipi_dsi/dsi_stm32.c:
```diff
@@ -504,24 +504,11 @@ static int mipi_dsi_stm32_init(const struct device *dev)
-			.bus = DT_INST_CLOCKS_CELL_BY_NAME(inst, pixelclk, bus),		\
-		},										\
+		.dsi_clk = STM32_DT_INST_CLOCK_INFO_BY_NAME(inst, dsiclk),			\
+		IF_ENABLED(DT_INST_CLOCKS_HAS_NAME(inst, name),					\
```
Addressed.

etienne-lms commented on drivers/ethernet/eth_dwmac_stm32h7x.c:
```diff
@@ -32,18 +32,11 @@ PINCTRL_DT_INST_DEFINE(0);
+
+static const struct stm32_pclken pclken = STM32_DWMAC_CLK(stm_eth);
+static const struct stm32_pclken pclken_tx = STM32_DWMAC_CLK(mac_clk_tx);
+static const struct stm32_pclken pclken_rx = STM32_DWMAC_CLK(mac_clk_rx);
```
Addressed.

mathieuchopstm commented:
First two commits can be amended to remove the `zephyr:` in `include: zephyr: drivers` too.
Also, I would `s/tree-wide/*` in `drivers: stm32 tree-wide: use STM32_CLOCK_INFO_BY_NAME and friends`.

Otherwise LGTM

etienne-lms commented on drivers/gpio/gpio_stm32.c:
```diff
@@ -736,16 +736,16 @@ __maybe_unused static int gpio_stm32_init(const struct device *dev)
 		.common = {						       \
 			 .port_pin_mask = GPIO_PORT_PIN_MASK_FROM_NGPIOS(16U), \
 		},							       \
 		.base = (uint32_t *)__base_addr,				       \
```
You and GH are right.

etienne-lms committed:
drivers: gpio: stm32: clean instance init macro indentation

Cleanup indentation in STM32 GPIO controllers device definition
macros. Replace some uses of COND_CODE_1() with IF_ENABLED() that
is bit more explicit.

Signed-off-by: Etienne Carriere <etienne.carriere@st.com>

etienne-lms commented:
I've changed the first commits message header lines
and appended commit *"drivers: gpio: stm32: clean instance init macro indentation"* to address https://github.com/zephyrproject-rtos/zephyr/pull/98995#discussion_r2524127319.
